core/trie/trie_test.go

```diff
@@ -87,52 +89,292 @@ func TestFindCommonPath(t *testing.T) {
+				assert.NoError(t, err)
+				assert.Equal(t, trie.FeltToBitSet(leftKey), parentNode.Left)
+				assert.Equal(t, trie.FeltToBitSet(newKey), parentNode.Right)
+			})
+			t.Run("Add to right branch", func(t *testing.T) {
+				newKeyNum, err := strconv.ParseUint("110", 2, 64)
+				require.NoError(t, err)
+				newKey := new(felt.Felt).SetUint64(newKeyNum)
+				newVal := new(felt.Felt).SetUint64(12)
+				_, err = trie.Put(newKey, newVal)
+				assert.NoError(t, err)
+				commonKey := bitset.New(251 - 1).Set(0).Set(1)
+				parentNode, err := trie.storage.Get(commonKey)
+				assert.NoError(t, err)
+				assert.Equal(t, trie.FeltToBitSet(newKey), parentNode.Left)
+				assert.Equal(t, trie.FeltToBitSet(rightKey), parentNode.Right)
+			})
+			t.Run("Add new node as parent sibling", func(t *testing.T) {
+				newKeyNum, err := strconv.ParseUint("000", 2, 64)
+				require.NoError(t, err)
+				newKey := new(felt.Felt).SetUint64(newKeyNum)
+				newVal := new(felt.Felt).SetUint64(12)
+				_, err = trie.Put(newKey, newVal)
+				assert.NoError(t, err)
+				commonKey := bitset.New(251 - 3)
+				parentNode, err := trie.storage.Get(commonKey)
+				assert.NoError(t, err)
+				assert.Equal(t, trie.FeltToBitSet(newKey), parentNode.Left)
+				expectRightKey := bitset.New(251 - 2).Set(0)
+				assert.Equal(t, expectRightKey, parentNode.Right)
+			})
+			return nil
+		})
+	})
+}
+
+func TestTrieDeleteBasic(t *testing.T) {
+	// left branch
+	leftKeyNum, err := strconv.ParseUint("100", 2, 64)
+	require.NoError(t, err)
+	leftKey := new(felt.Felt).SetUint64(leftKeyNum)
+	leftVal := new(felt.Felt).SetUint64(12)
+	// right branch
+	rightKeyNum, err := strconv.ParseUint("111", 2, 64)
+	require.NoError(t, err)
+	rightKey := new(felt.Felt).SetUint64(rightKeyNum)
+	rightVal := new(felt.Felt).SetUint64(22)
+	// Zero value
+	zeroVal := new(felt.Felt).SetUint64(0)
 	tests := [...]struct {
-		key   *felt.Felt
-		value *felt.Felt
-		root  *bitset.BitSet
+		deleteKeys []*felt.Felt
+		assertFun  func(trie *Trie)
 	}{
 		{
-			key:   new(felt.Felt).SetUint64(2),
-			value: new(felt.Felt).SetUint64(2),
-			root:  nil,
+			deleteKeys: []*felt.Felt{leftKey},
+			assertFun: func(trie *Trie) {
+				assert.Equal(t, trie.FeltToBitSet(rightKey), trie.rootKey)
+				val, err := trie.Get(leftKey)
+				assert.Nil(t, val)
+				assert.Error(t, err)
+			},
 		},
 		{
-			key:   new(felt.Felt).SetUint64(1),
-			value: new(felt.Felt).SetUint64(1),
-			root:  nil,
+			deleteKeys: []*felt.Felt{rightKey},
+			assertFun: func(trie *Trie) {
+				assert.Equal(t, trie.FeltToBitSet(leftKey), trie.rootKey)
+				val, err := trie.Get(rightKey)
+				assert.Nil(t, val)
+				assert.Error(t, err)
+			},
 		},
 		{
-			key:   new(felt.Felt).SetUint64(3),
-			value: new(felt.Felt).SetUint64(3),
-			root:  nil,
+			deleteKeys: []*felt.Felt{leftKey, rightKey},
+			assertFun: func(trie *Trie) {
+				assert.Nil(t, trie.rootKey)
+			},
 		},
+	}
+	for testIdx, test := range tests {
+		RunOnTempTrie(251, func(trie *Trie) error {
```

Comment:
Since there are running on separate tries you should use subtests using `t.Run()` as follows:

```go
for testIdx, test := range tests {
  t.Run(test.name, func(t *testing){...}
}
```

Based on the above test cases the names can be "delete right key", "delete left key" and "delete left and right keys"